ZipArchive.cs

```diff
@@ -255,13 +440,16 @@ string NormalizeArchivePath (bool isDir, string archivePath)
 
 			if (isDir) {
 				if (archivePath [archivePath.Length - 1] != Path.DirectorySeparatorChar)
-					return archivePath + Path.DirectorySeparatorChar;
-				return archivePath;
+					archivePath = archivePath + Path.DirectorySeparatorChar;
+			}
+			else if (archivePath [archivePath.Length - 1] == Path.DirectorySeparatorChar)
+					archivePath = archivePath.Substring (0, archivePath.Length - 1);
+
+			if (Path.IsPathRooted (archivePath)) {
+				archivePath = archivePath.Remove (0, Path.GetPathRoot (archivePath).Length);
 			}
 
-			if (archivePath [archivePath.Length - 1] == Path.DirectorySeparatorChar)
-				return archivePath.Substring (0, archivePath.Length - 1);
-			return archivePath;
+			return archivePath.Replace ("\\", "/");
```

Comment:
We kinda need this because users might give us pass data with "\" rather than "/"